app/controllers/api/v1x0/portfolio_items_controller.rb

```diff
@@ -9,6 +9,8 @@ def index
         else
           collection(PortfolioItem.all)
         end
+      rescue ActionController::ParameterMissing => e
```

Comment:
@lindgrenj6 do we need to rescue this I think rails sends out the correct message when it catches this exception
This is a good read on the internals of rails and exceptions.
https://ieftimov.com/post/how-rails-handles-status-codes/

From a rails console if you run this 
ActionDispatch::ExceptionWrapper.rescue_responses

You will see the http status codes that rails handles internally when it traps an exception and it needs to report back to the client

Currently rails does this as
"ActionController::ParameterMissing"=>:bad_request

Are we trying to change this behavior

Reply:
@lindgrenj6 
404 versus 422
https://www.bennadel.com/blog/2434-http-status-codes-for-invalid-data-400-vs-422.htm

If we want to use 422 for that reason I understand but would it make more sense to change the ActionDispatch::ExceptionWrapper.rescue_responses in one of our config locations so its available generically to all controllers and we won't have to change every file.

Reply:
@lindgrenj6 
Maybe if we add a file config/initializers/http_rescue_responses.rb with this line
```
ActionDispatch::ExceptionWrapper.rescue_responses["ActionController::ParameterMissing"] = :unprocessable_entity
```
It might do the trick for us in one place

Reply:
I implemented this - and the next problem we're running into is the fact that we're swallowing `KeyError` here https://github.com/ManageIQ/catalog-api/blob/6123893880c14260bd9571cae9caf044bc1cedaa/app/controllers/application_controller.rb#L18-L19

`ActionController::ParameterMissing` extends `KeyError`, so it gets caught here. Would you be ok with adding a specific rescue for it?

Reply:
@lindgrenj6 @syncrou 
The KeyError is our synthetic error coming if someone accidentally calls us with a bad user object
Which should emanate from here
https://github.com/ManageIQ/manageiq-api-common/blob/master/lib/manageiq/api/common/request.rb
And the properties on the UserObject
https://github.com/ManageIQ/manageiq-api-common/blob/e2f0c27d6f48df5a1c477a6d06a94832c8da3caf/lib/manageiq/api/common/user.rb#L36
Thats being called as an identity error.

Since most of this data is inserted by 3scale when a user logs in, its unlike that the user sent bad data. We should account for a bad header coming in but we should not be using the KeyError but extend the Identity Error to say that the x-rh-identity is invalid or incompatible with code.

The other story behind this is we used to use the x-rh-identity to do some pre RBAC validation called UserConstraints and AdminConstraints. @syncrou might know about this. We don't use the 2 constraints anymore so we should be able to remove this exception.